### DISCUSSION TITLE
[Feature] Tinder-Like Junk Photo Cleaner for Immich

### DISCUSSION THREAD

**Discussion by encodedLife** [immich-app/immich#9822] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

First of all, I would like to commend the developers for the fantastic work done on the Immich Photomanager so far. Your efforts have created an excellent tool for photo management. 

I previously used an Android app called "Junk Photo Cleaner" ([Junk Photo Cleaner](https://play.google.com/store/apps/details?id=com.stash.junkcleaner)). This app was outstanding, but it eventually became unusable. The app scanned my gallery and generated a list of photos identified by its AI as "trash." I could then review these marked photos, similar to a Tinder-like interface, and choose which ones to keep and which to discard.

I believe adding a similar feature to Immich Photomanager would set it apart from other photo management tools. I have yet to find a similar function in any other photo management software. With the increasing number of photos and videos we accumulate, many of them end up being unnecessary and waste valuable storage space. An AI-powered feature in Immich could efficiently identify potentially trash photos and suggest them for deletion, helping users manage their storage more effectively.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
I've added this as a comment on #7202

**Reply by encodedLife** [reply]
Hi bo0tzz,

Thanks for pointing me to the discussion on #7202. I appreciate the connection and see the potential overlap.

To expand on the idea, I believe integrating an AI-driven junk photo cleaner that functions similarly to a Tinder-like interface would offer a unique user experience. While #7202 discusses using an ML model for ranking and selecting high-quality images, my suggestion focuses on identifying and proposing low-quality or unwanted photos for deletion.

Here's how it could complement the existing discussion:

- AI-Powered Trash Detection: Use a machine learning model to identify photos that are likely to be "trash" based on quality, duplicates, or other criteria.

- Tinder-Like Interface: Present these identified photos in a swipeable interface, allowing users to quickly decide which photos to keep or delete. This interactive approach can make the cleanup process more engaging and efficient.

- Storage Management: By effectively identifying and removing junk photos, users can reclaim valuable storage space, making their photo library more manageable.

Combining this with the ideas in #7202, we could create a comprehensive photo management tool that not only highlights the best photos but also helps users maintain a clutter-free library.

Looking forward to any thoughts or feedback on this expanded concept.

**Comment by MaaxGr** [comment]
Hey @encodedLife,
i liked the idea and vibe coded a prototype today. Let me know what you think :)
https://github.com/MaaxGr/ImmichTinder

I will use it to fill an album with pictures i like and display that album on my google tv.

**Comment by t0msk** [comment]
Why this feature is closed? I would like to see this, you could launch Tinder for any Album and if you swipe for yes it will keep and if you swipe for no it will move to trash, where it will be deleted after 30 days.

**Comment by iddu0** [comment]
I'm working on a big project right now, if I have time, I may try to make a React Native "Immich Tinder" app (like Swipewipe, but FOSS!)

**Comment by encodedLife** [comment]
Hi everyone, first of all, thanks a lot for the discussion around this idea. And @MaaxGr, thanks for your MVP as well. It already looks very promising.

What I personally would lean towards is integrating this directly into the official Immich mobile app as a native cleanup and culling mode, instead of having it only as an external tool.

One important clarification: I do not think this feature should ever automatically delete photos. It should only help with ranking and selecting the photos that are shown to the user during review. The user would still manually decide what goes to trash. After that, Immich’s existing trash behavior can handle deletion as usual, for example deleting trashed items after the configured retention period.

So the goal would not be:

```text
AI decides what should be deleted
```

but rather:

```text
Immich shows likely cleanup candidates first, based on previous user decisions
```

Technically, I would also avoid training a separate AI model for this. The existing image embedding model could stay completely unchanged and only be used as a frozen feature extractor. For example, Immich could use the existing smart search or CLIP-like image embeddings and combine them with metadata or simple quality signals.

For every asset, Immich could build a feature vector like:

```text
features(asset) = [
  image_embedding,
  blur_score,
  exposure_score,
  duplicate_score,
  screenshot_or_document_flags,
  has_detected_people,
  is_favorite,
  album_count,
  asset_type,
  created_at_or_import_context,
  camera_or_EXIF_signals
]
```

The mobile app would then show a swipe-based cleanup queue. The user could mark each item as:

```text
delete / keep / archive / skip
```

These user actions become labels:

```text
(asset_id, user_id, feature_vector, label)
```

Instead of retraining the neural network, Immich would only train a lightweight per-user classifier on top of the existing embeddings and metadata.

For example:

```text
Logistic Regression
Linear SVM
kNN over embeddings
or a simple centroid-based similarity classifier
```

With logistic regression, the classifier could output:

```text
p_delete = sigmoid(w · features(asset) + b)
```

This would not mean “delete this photo”. It would only mean:

```text
This photo looks similar to photos this user has previously moved to trash.
```

The next cleanup session could then work like this:

```text
1. Load all unreviewed / ToReview assets.
2. Generate or load their existing embeddings and metadata features.
3. Load the user-specific lightweight classifier.
4. Score each asset with p_delete.
5. Apply protection rules.
6. Sort the review queue by score.
7. Show the highest-ranked candidates first in the mobile app.
```

The classifier could be updated periodically, for example:

```text
after a cleanup session ends
after N new labels have been collected
once per night as a background job
```

I would not retrain anything after every single swipe, and I would definitely not retrain the actual neural network. The neural network remains generic. The classifier only learns the cleanup preference of a specific user.

Protection rules would still be important. Assets should be downranked or excluded if they are favorites, part of albums, shared, contain known people, are RAW files, were recently uploaded, or were previously marked as keep.

In my opinion, this implementation should actually be fairly quick to build compared to a completely new AI feature, because Immich already has the most complex parts in place: asset metadata, thumbnails, trash handling, mobile apps, machine learning infrastructure, image embeddings, and duplicate review concepts.

This would mainly add a personalized sorting and ranking layer for the mobile review queue. It helps the user find likely cleanup candidates faster, while all destructive decisions remain manual and go through Immich’s existing trash flow.

I also think this could be a feature that clearly sets Immich apart from many other photo management apps. Most apps are good at storing photos, but not many are good at helping users clean up their library in a safe, personalized, privacy-respecting way.
